specs/core/1_new_shards.md

```diff
@@ -0,0 +1,583 @@
+
+```python
+def get_light_client_committee(beacon_state: BeaconState, epoch: Epoch) -> Sequence[ValidatorIndex]:
+    source_epoch = epoch - epoch % LIGHT_CLIENT_COMMITTEE_PERIOD 
+    if source_epoch > 0:
+        source_epoch -= LIGHT_CLIENT_COMMITTEE_PERIOD
+    active_validator_indices = get_active_validator_indices(beacon_state, source_epoch)
+    seed = get_seed(beacon_state, source_epoch, DOMAIN_SHARD_LIGHT_CLIENT)
+    return compute_committee(active_validator_indices, seed, 0, ACTIVE_SHARDS)[:TARGET_COMMITTEE_SIZE]
+```
+
+### `get_indexed_attestation`
+
+```python
+def get_indexed_attestation(beacon_state: BeaconState, attestation: Attestation) -> AttestationAndCommittee:
+    committee = get_beacon_committee(beacon_state, attestation.data.slot, attestation.data.index)
+    return AttestationAndCommittee(committee, attestation)
+```
+
+### `get_updated_gasprice`
+
+```python
+def get_updated_gasprice(prev_gasprice: Gwei, length: uint8) -> Gwei:
+    if length > BLOCK_SIZE_TARGET:
+        delta = prev_gasprice * (length - BLOCK_SIZE_TARGET) // BLOCK_SIZE_TARGET // GASPRICE_ADJUSTMENT_COEFFICIENT
+        return min(prev_gasprice + delta, MAX_GASPRICE)
+    else:
+        delta = prev_gasprice * (BLOCK_SIZE_TARGET - length) // BLOCK_SIZE_TARGET // GASPRICE_ADJUSTMENT_COEFFICIENT
+        return max(prev_gasprice, MIN_GASPRICE + delta) - delta
+```
+
+### `is_valid_indexed_attestation`
+
+```python
+def is_valid_indexed_attestation(state: BeaconState, indexed_attestation: AttestationAndCommittee) -> bool:
+    """
+    Check if ``indexed_attestation`` has valid indices and signature.
+    """
+
+    # Verify aggregate signature
+    all_pubkeys = []
+    all_message_hashes = []
+    aggregation_bits = indexed_attestation.attestation.aggregation_bits
+    assert len(aggregation_bits) == len(indexed_attestation.committee)
+    for i, custody_bits in enumerate(indexed_attestation.attestation.custody_bits):
+        assert len(custody_bits) == len(indexed_attestation.committee)
+        for participant, abit, cbit in zip(indexed_attestation.committee, aggregation_bits, custody_bits):
+            if abit:
+                all_pubkeys.append(state.validators[participant].pubkey)
+                # Note: only 2N distinct message hashes
+                all_message_hashes.append(hash_tree_root(
+                    AttestationCustodyBitWrapper(hash_tree_root(indexed_attestation.data), i, cbit)
+                ))
+            else:
+                assert cbit == False
+        
+    return bls_verify_multiple(
+        pubkeys=all_pubkeys,
+        message_hashes=all_message_hashes,
+        signature=indexed_attestation.signature,
+        domain=get_domain(state, DOMAIN_BEACON_ATTESTER, indexed_attestation.data.target.epoch),
+    )
+```
+
+### `get_shard`
+
+```python
+def get_shard(state: BeaconState, attestation: Attestation) -> Shard:
+    return Shard((attestation.data.index + get_start_shard(state, data.slot)) % ACTIVE_SHARDS)
```

Comment:
`get_start_shard` is no longer defined in beacon chain spec, I think we have to reintroduce it here